### ISSUE TITLE
Consumes  */*  in OpenAPI breaks Swagger UI

### ISSUE THREAD

**Issue by zerodayz** [swagger-api/swagger-ui#6742]
Consumes set to */* in OpenAPI spec, breaks swagger ui request.

For example in Kubernetes:

curl -X DELETE "HOST/api/v1/namespaces/test-namespace/pods/test-pod?propagationPolicy=Background" -H  "accept: application/json" -H  "Content-Type: */*" -d "{  \"propagationPolicy\": \"Background\"}"

breaks in app because it takes */* literally, instead you should be able to choose from:
 - application/json
 - application/yaml
 - application/vnd.kubernetes.protobuf

Such curl requests will results in:

  "message": "the body of the request was in an unknown format - accepted media types include: application/json, application/yaml, application/vnd.kubernetes.protobuf",
  "reason": "UnsupportedMediaType",

Is there a way to add custom content-types when */* is matched?

**Comment by zerodayz** [comment]
Workaround using injector:

```
    requestInterceptor: function(req) {
      req.headers['Content-Type'] = 'application/json'
      return req
  }
```

**Comment by zerodayz** [comment]
It would be great if its possible to set the header to application/json if consumes is set to */* only. As the workaround breaks other APIs that have proper consumes.

**Comment by hkosova** [comment]
Please post your OpenAPI YAML/JSON file.
Also, what version of Swagger UI do you use? (check the `versions` value in the browser dev tools)

**Comment by zerodayz** [comment]
@hkosova I am using Kubernetes API: https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.20/api/openapi-spec/swagger.json

Problem occurs with any API that has `consumes: */*`
 ```
   "/api/v1/namespaces/{namespace}/events/{name}": {
      "delete": {
        "consumes": [
          "*/*"
        ],
```

Swagger UI curl requests automatically adds: curl -H "Content-Type: */*" which by the API is rejected as Unknown Media Type.

**Comment by zerodayz** [comment]
I am using master swagger-ui from https://github.com/swagger-api/swagger-ui, cloned today.

**Comment by hkosova** [comment]
The problem is with Kubernetes OpenAPI definition. If their API only accepts request bodies with specific media types, such as

> accepted media types include: application/json, application/yaml, application/vnd.kubernetes.protobuf",

then `consumes` must specify these exact media types instead of the `*/*` wildcard:

```json
        "consumes": [
          "application/json",
          "application/yaml",
          "application/vnd.kubernetes.protobuf",
        ],
```

**Comment by zerodayz** [comment]
However 'consumes: */*' is valid OpenAPI definition

**Comment by hkosova** [comment]
Yes - provided that the API server actually accepts `Content-Type: */*`. The API definition should match the implementation.

**Comment by zerodayz** [comment]
But doesnt wildcard just represent anytype? for example if I send the application/json it works of course, so type wildcard works, the problem is that Swagger UI sent literally wildcard. Which I think is incorrect. In such situations at best I should be able to definy type myself as Server accepts wildcard.
